### DISCUSSION TITLE
SSLError - Is it my system's security level?

### DISCUSSION THREAD

**Discussion by ac1987** [theOehrly/Fast-F1#405] [Q&A] [answered]
I can successfully run the following code on a personal computer on a home network. However, I receive an error when trying to run on an old work computer on the same network. 

`from fastf1.ergast import Ergast`
`ergast = Ergast(result_type='pandas', auto_cast=True)`
`result = ergast.get_race_results(season=2022)`

Error:
`SSLError: HTTPSConnectionPool(host='ergast.com', port=443): Max retries exceeded with url: /api/f1/2022/results.json (Caused by SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate (_ssl.c:997)')))`

When working directly through the Ergast API, I successfully retrieve data by setting `verify=False` in `request.get(url, verify=False).json.` This leads me to believe I have some issue with the security level imposed on the system. Does anyone know of quick fix to either circumvent the security or reduce the level of security to avoid these errors and download the data? 
**_Thanks!_**

**Comment by theOehrly** [comment, marked as answer]
@ac1987 I can't really tell what the root issue here is. Most likely the machine is missing one of the required root certificates. Either because it is an old OS maybe. Or because of some IT security policy. (Or an old version of Python?)

In any case, you should be able to use the HTTP endpoint of Ergast instead of HTTPS by overwriting this at the beginning of every script that you run.

Example (not tested, but I think this should work):

```python
from fastf1.ergast import interface
interface.BASE_URL = 'http://ergast.com/api/f1'

.... your other code ....
```

**Comment by ac1987** [comment]
@theOehrly Many thanks! That worked! 

I am using Python 3.10.5. on Windows10. I am leaning to it being the IT security policy.

Applying this method to the livetiming url has appeared to have fixed the other issue I was encountering while trying to load and cache data via fastf1.

`from fastf1 import _api`
`_api.base_url = 'http://livetiming.formula1.com`

Any idea why a security policy would cause my system to not have the required root certificates?

Thanks again!

**Reply by theOehrly** [reply]
Sorry, I have absolutely no idea why this is happening for you. It being some security policy thing is just a guess, but there could be other reasons. I especially have no idea why it's happening with both URLs.

You could try to install https://pypi.org/project/pip-system-certs/ and see if that fixes the issue. You're company might use a firewall that does MITM HTTPS traffic scanning and your PC has a self signed root cert installed to trust this traffic. Python ships with its own certs though, but this package patches it to use the system certs.
